src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -601,20 +344,183 @@ void CardTableRS::verify() {
 CardTableRS::CardTableRS(MemRegion whole_heap) :
   CardTable(whole_heap) { }
 
+// Implemented word-iteration to skip long consecutive clean cards.
+CardTable::CardValue* CardTableRS::find_first_dirty_card(CardValue* const start_card,
+                                                         CardValue* const end_card) {
+  using Word = uintptr_t;
+
+  auto is_word_aligned = [] (const void* const p) {
```

Comment:
Just my opinion, feel free to ignore: don't use `auto` if unavoidable, in this case use the specific type `bool`.